### PR TITLE
swap_legs_with_ids

### PR DESCRIPTION
**Pull request by Omar-Oz**

### CODE REVIEW

**Comment by steco** [comment]
Is there a jira for this so I can understand the requirements?

File: examples/use-cases/ibor/Example of a Stock Acquisition.ipynb [hunk 1]
```diff
@@ -920,4 +920,4 @@
  },
  "nbformat": 4,
  "nbformat_minor": 5
-}
```
Comment (steco) [inline examples/use-cases/ibor/Example of a Stock Acquisition.ipynb:923]: Wanna revert this change?

File: examples/use-cases/instruments/Swap upesert with leg id's.ipynb [hunk 2]
```diff
@@ -0,0 +1,476 @@
+{
```
Comment (steco) [inline examples/use-cases/instruments/Swap upesert with leg id's.ipynb:1]: The name of this file has a typo - "upesert"

Comment (steco) [inline examples/use-cases/instruments/Swap upesert with leg id's.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    ## Define flow convention for Fixed LEg
</pre><p>"Leg" </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/460/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (steco) [inline examples/use-cases/instruments/Swap upesert with leg id's.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #79.    irs_legs = [
</pre><p>Might be simpler to include this?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/460/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (steco) [inline examples/use-cases/instruments/Swap upesert with leg id's.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #15.            create_property_definition_request=property_definition_request
</pre><p>Inline this?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/460/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (steco) [inline examples/use-cases/instruments/Swap upesert with leg id's.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #55.    IRS_definition = models.InstrumentDefinition(
</pre><p>Suggest you move this line to the full instrument definition section above.</p><p> </p><p>Also slightly confusing that this upserts both swaps, perhaps split into the appropriate sections?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/460/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

**Comment by steco** [comment]
I'm confused by this notebook - does it demonstrate how to add swapleg ids to a full instrument economic definition, or a separate instrument?

File: examples/use-cases/instruments/Swap upesert with leg id's.ipynb (hunk 2 above)
Comment (rizwansaeed) [inline examples/use-cases/instruments/Swap upesert with leg id's.ipynb:1]: <p>use the template https://github.com/finbourne/sample-notebooks/blob/master/docs/template.ipynb so we get consistent package names and formatting</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/460/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (rizwansaeed) [inline examples/use-cases/instruments/Swap upesert with leg id's.ipynb:1]: <p>Add some commentary (in markdown) explaining what is happening, or link to an appropriate KB article if it exists</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/460/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (rizwansaeed) [inline examples/use-cases/instruments/Swap upesert with leg id's.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #28.    # ccy/tenor/indexname dependency
</pre><p> </p><p>is this describing the <code>fixing_reference</code> field? if so put the comment in the same format </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/460/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (rizwansaeed) [inline examples/use-cases/instruments/Swap upesert with leg id's.ipynb:1]: <p>Formatting</p><p> </p><p>This looks like its a very specific use case (which is fine), add some commentary to explain what is happening i.e. creating properties isn't <em>required</em> but is a nice feature. Also update the intro description </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/460/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (rizwansaeed) [inline examples/use-cases/instruments/Swap upesert with leg id's.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #17.        property_definition_request2 = models.CreatePropertyDefinitionRequest(
</pre><p>give this a more appropriate name for clarity e.g. <code>leg_id_request</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/460/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (rizwansaeed) [inline examples/use-cases/instruments/Swap upesert with leg id's.ipynb:1]: <p>this needs some explanation as to why you are doing this</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/460/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (rizwansaeed) [inline examples/use-cases/instruments/Swap upesert with leg id's.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #13.                #unit="Decimal",
</pre><p>remove the commented out values</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/460/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (rizwansaeed) [inline examples/use-cases/instruments/Swap upesert with leg id's.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    print(listofinstruments.values[0].properties)
</pre><p>format this as a data frame and use <code>display</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/460/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (rizwansaeed) [inline examples/use-cases/instruments/Swap upesert with leg id's.ipynb:1]: <p>format this as a data frame and use&nbsp;<code>display</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/460/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>
